### ISSUE TITLE
About how to generate surface normal vector maps in real time

### ISSUE THREAD

**Issue by ShiLong0526** [chaytonmin/Off-Road-Freespace-Detection#16]
Hi! Are the 3 demo images you guys added to the readme processed in real time in an in-car environment? If so, how were the surface normal vector maps generated in real time? I use ROS(Robot Operating System) to read the depth map topic to calculate the surface normal vector maps using SNE and can't do it to generate the surface normal vector maps in real time. Please let me know if there is a better way, thanks!

**Comment by chaytonmin** [comment]
> normal

Yes, currently generating normals takes a long time and cannot be done in real-time. The demo presented is a video stitched together from the results. However, it should be possible to improve the SNE code to accelerate normal generation.
